CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+    
+    -   If your PR cannot be merged automatically, please resolve merge conflicts on your development branch and resubmit the PR.
+
+4.  Review process:
+
+    -   Reviewers (the other developers and interested community members) will write inline and/or general comments on your Pull Request (PR) to help you improve its implementation, documentation and style. We're all very friendly and open to learning from each other while improving the code, so don't let the review process discourage you from contributing! Also, if your PR is not getting any reaction, feel free to email the developer team at ev-help@caltech.edu.
+    -   To update your PR, make your changes on your local repository, commit, run tests, and **only if they succeed** push to your fork. As soon as those changes are pushed up (to the same branch as before) the PR will update automatically. If you have no idea how to fix the test failures, you may push your changes anyway and ask for help in a PR comment.
+    -   Various continuous integration (CI) services are triggered after each PR update to build the code, run unit tests, and check coding style of your branch. The CI tests should pass before your PR can be merged. If CI fails, you can find out why by clicking on the “failed” icon (red cross) and inspecting the build and test log. To avoid overuse and waste of this resource, test your work locally before committing. If you can't figure out why one or more steps of the CI pipeline is failing, you can ask for help in a PR comment.
+    -   A PR must be approved by at least one core team member before merging. Approval means the core team member has carefully reviewed the changes, and the PR is ready for merging. A member of the team will give the LGTM and you may then merge if a member of the team has not already done so.
+
+5.  User-facing changes: when a PR makes user-facing changes, we have some general guidelines we like to follow:
+
+    -   Avoid changing user-facing function signatures/names. If absolutely necessary, add arguments using kwargs with defaults; "remove" arguments by deprecating the argument when input. If changing a function or attribute name, add a deprecated access to the old function/attribute name.
+    -   You may need to add additional files/modules to the docs so that readthedocs finds the new docstrings.
+    -   New dependencies should be avoided if possible. Depending on the importance of the feature and the activity of the new dependency, the core team may overrule this, but such should be discussed on Github.
+    -   If you are proposing a change that affects multiple packages of the ACN Research Portal, please contact the developer team at sunash.sharma@gmail.com to discuss.
+
+## Stylistic Guidelines
```

Comment:
I started using black (https://github.com/psf/black) on PowerFlex work. The goal of these auto formatters is to make style consistent and avoid dealing with style in code reviews. What do you think?

I also tried out YAPF (https://github.com/google/yapf) which is more flexible and I think I like the style options better, but some people complain it is not deterministic.

Reply:
I'm leaning towards using black because of the determinism. Also, the lack of config options does decrease the number of decisions we have to make, which is nice. I agree that YAPF has nicer-looking styles but I think black would in general give us fewer maintenance headaches in the future. That black checks for an unchanged AST is also comforting. I can run black over all of acnportal in this PR if we want to just get that out of the way.

Reply:
Also, if black and Codacy disagree, we can remove codacy and add a black --check or similar to the Travis build (i.e. check if the black formatter has been run on the Python code as part of every PR).

Reply:
I like some of the other checks in codacy, so if possible I'd like to keep both, but if it becomes a bother we can drop it.